### PR TITLE
PROD-769 Use role=creator syntax

### PR DESCRIPTION
**Pull request by Qi77Qi**
There’s a bug currently where users might see a runtime they don’t own in analysis page, and when they try to start it, they’ll get permission error (becuz it’s a runtime created by someone else)…this is only an issue for workspace owners

This is due to listRuntime call is using the old `creator=<email>` syntax, which should still be supported by Leo api. But we missed the fact that when the request is sent from UI, the `@` is encoded as `%40`. Hence the creator filter isn't being used properly

### CODE REVIEW

**Review by n8landolt — APPROVED** [review]

**Review by mlilynolting — APPROVED** [review]

**Review by n8landolt — DISMISSED** [review]

File: src/pages/workspaces/workspace/WorkspaceContainer.js
Comment (cahrens) [inline src/pages/workspaces/workspace/WorkspaceContainer.js]: What about the usage at line 296? 
https://github.com/DataBiosphere/terra-ui/blob/1fea8ec729bb13da97dd5321b34292d46c24703c/src/pages/workspaces/workspace/WorkspaceContainer.js#L296

**Comment by cahrens** [comment]
Is the disk API impacted (https://github.com/DataBiosphere/terra-ui/blob/cbb67231a3756a18a609531a4d9e040289d25a11/src/pages/workspaces/List.js#L158)?

**Comment by Qi77Qi** [comment]
so I did a bit more testing, and seems like leo is accepting filter like `creator=qiwangterra%40gmail.com`...but if tests succeed, I'm still gonna try to push this PR through just to see if it makes a difference

**Comment by cahrens** [comment]
Please do include information about the testing that was done with the complete set of changes (after you decide about https://github.com/DataBiosphere/terra-ui/pull/3643#issuecomment-1372343018). I'm a bit nervous because approvals from approvals from @n8landolt and @mlilynolting were from a partial set of changes.

**Comment by Qi77Qi** [comment]
```
➜  Github curl 'https://notebooks.firecloud.org/api/v2/runtimes?saturnAutoCreated=true&creator=qi%40broadinstitute.org&saturnWorkspaceName=qi-new-12-14&saturnWorkspaceNamespace=qi-new-12-14' \
  -H "Authorization: Bearer `gcloud auth print-access-token`"
[{"id":71872,"workspaceId":null,"runtimeName":"saturn-f0cbf654-4d0d-4dd6-aa93-562f0cf66f21","googleProject":"terra-07381aa8","cloudContext":{"cloudProvider":"GCP","cloudResource":"terra-07381aa8"},"auditInfo":{"creator":"qi@broadinstitute.org","createdDate":"2022-12-22T14:43:14.899406Z","destroyedDate":null,"dateAccessed":"2022-12-22T15:19:56.915675Z"},"runtimeConfig":{"numberOfWorkers":0,"masterMachineType":"n1-standard-4","masterDiskSize":150,"workerMachineType":null,"workerDiskSize":null,"numberOfWorkerLocalSSDs":null,"numberOfPreemptibleWorkers":null,"cloudService":"DATAPROC","region":"us-central1","componentGatewayEnabled":true,"workerPrivateAccess":false},"proxyUrl":"https://notebooks.firecloud.org/proxy/terra-07381aa8/saturn-f0cbf654-4d0d-4dd6-aa93-562f0cf66f21/jupyter","status":"Stopped","labels":{"saturnWorkspaceNamespace":"qi-new-12-14","saturn-iframe-extension":"https://app.terra.bio/jupyter-iframe-extension.js","creator":"qi@broadinstitute.org","clusterServiceAccount":"pet-114543224357351345371@terra-07381aa8.iam.gserviceaccount.com","saturnAutoCreated":"true","clusterName":"saturn-f0cbf654-4d0d-4dd6-aa93-562f0cf66f21","saturnWorkspaceName":"qi-new-12-14","saturnVersion":"6","tool":"Jupyter","runtimeName":"saturn-f0cbf654-4d0d-4dd6-aa93-562f0cf66f21","cloudContext":"Gcp/terra-07381aa8","googleProject":"terra-07381aa8"},"patchInProgress":false},{"id":72404,"workspaceId":null,"runtimeName":"saturn-02636156-af51-4ca9-8cd8-c03a196c6fc7","googleProject":"terra-07381aa8","cloudContext":{"cloudProvider":"GCP","cloudResource":"terra-07381aa8"},"auditInfo":{"creator":"mcnatt@broadinstitute.org","createdDate":"2023-01-03T22:18:44.141960Z","destroyedDate":null,"dateAccessed":"2023-01-03T22:54:06.258843Z"},"runtimeConfig":{"machineType":"n1-standard-1","persistentDiskId":18662,"cloudService":"GCE","bootDiskSize":120,"zone":"us-central1-a","gpuConfig":null},"proxyUrl":"https://notebooks.firecloud.org/proxy/terra-07381aa8/saturn-02636156-af51-4ca9-8cd8-c03a196c6fc7/jupyter","status":"Stopped","labels":{"saturnWorkspaceNamespace":"qi-new-12-14","saturn-iframe-extension":"https://app.terra.bio/jupyter-iframe-extension.js","creator":"mcnatt@broadinstitute.org","clusterServiceAccount":"pet-266964691988800639cb9@terra-07381aa8.iam.gserviceaccount.com","saturnAutoCreated":"true","clusterName":"saturn-02636156-af51-4ca9-8cd8-c03a196c6fc7","saturnWorkspaceName":"qi-new-12-14","saturnVersion":"6","tool":"Jupyter","runtimeName":"saturn-02636156-af51-4ca9-8cd8-c03a196c6fc7","cloudContext":"Gcp/terra-07381aa8","googleProject":"terra-07381aa8"},"patchInProgress":false}]%
➜  Github curl 'https://notebooks.firecloud.org/api/v2/runtimes?saturnAutoCreated=true&role=creator&saturnWorkspaceName=qi-new-12-14&saturnWorkspaceNamespace=qi-new-12-14' \
  -H "Authorization: Bearer `gcloud auth print-access-token`"
[{"id":71872,"workspaceId":null,"runtimeName":"saturn-f0cbf654-4d0d-4dd6-aa93-562f0cf66f21","googleProject":"terra-07381aa8","cloudContext":{"cloudProvider":"GCP","cloudResource":"terra-07381aa8"},"auditInfo":{"creator":"qi@broadinstitute.org","createdDate":"2022-12-22T14:43:14.899406Z","destroyedDate":null,"dateAccessed":"2022-12-22T15:19:56.915675Z"},"runtimeConfig":{"numberOfWorkers":0,"masterMachineType":"n1-standard-4","masterDiskSize":150,"workerMachineType":null,"workerDiskSize":null,"numberOfWorkerLocalSSDs":null,"numberOfPreemptibleWorkers":null,"cloudService":"DATAPROC","region":"us-central1","componentGatewayEnabled":true,"workerPrivateAccess":false},"proxyUrl":"https://notebooks.firecloud.org/proxy/terra-07381aa8/saturn-f0cbf654-4d0d-4dd6-aa93-562f0cf66f21/jupyter","status":"Stopped","labels":{"saturnWorkspaceNamespace":"qi-new-12-14","saturn-iframe-extension":"https://app.terra.bio/jupyter-iframe-extension.js","creator":"qi@broadinstitute.org","clusterServiceAccount":"pet-114543224357351345371@terra-07381aa8.iam.gserviceaccount.com","saturnAutoCreated":"true","clusterName":"saturn-f0cbf654-4d0d-4dd6-aa93-562f0cf66f21","saturnWorkspaceName":"qi-new-12-14","saturnVersion":"6","tool":"Jupyter","runtimeName":"saturn-f0cbf654-4d0d-4dd6-aa93-562f0cf66f21","cloudContext":"Gcp/terra-07381aa8","googleProject":"terra-07381aa8"},"patchInProgress":false}]%
➜  Github curl 'https://notebooks.firecloud.org/api/v2/runtimes?saturnAutoCreated=true&creator=qi@broadinstitute.org&saturnWorkspaceName=qi-new-12-14&saturnWorkspaceNamespace=qi-new-12-14' \
  -H "Authorization: Bearer `gcloud auth print-access-token`"
[{"id":71872,"workspaceId":null,"runtimeName":"saturn-f0cbf654-4d0d-4dd6-aa93-562f0cf66f21","googleProject":"terra-07381aa8","cloudContext":{"cloudProvider":"GCP","cloudResource":"terra-07381aa8"},"auditInfo":{"creator":"qi@broadinstitute.org","createdDate":"2022-12-22T14:43:14.899406Z","destroyedDate":null,"dateAccessed":"2022-12-22T15:19:56.915675Z"},"runtimeConfig":{"numberOfWorkers":0,"masterMachineType":"n1-standard-4","masterDiskSize":150,"workerMachineType":null,"workerDiskSize":null,"numberOfWorkerLocalSSDs":null,"numberOfPreemptibleWorkers":null,"cloudService":"DATAPROC","region":"us-central1","componentGatewayEnabled":true,"workerPrivateAccess":false},"proxyUrl":"https://notebooks.firecloud.org/proxy/terra-07381aa8/saturn-f0cbf654-4d0d-4dd6-aa93-562f0cf66f21/jupyter","status":"Stopped","labels":{"saturnWorkspaceNamespace":"qi-new-12-14","saturn-iframe-extension":"https://app.terra.bio/jupyter-iframe-extension.js","creator":"qi@broadinstitute.org","clusterServiceAccount":"pet-114543224357351345371@terra-07381aa8.iam.gserviceaccount.com","saturnAutoCreated":"true","clusterName":"saturn-f0cbf654-4d0d-4dd6-aa93-562f0cf66f21","saturnWorkspaceName":"qi-new-12-14","saturnVersion":"6","tool":"Jupyter","runtimeName":"saturn-f0cbf654-4d0d-4dd6-aa93-562f0cf66f21","cloudContext":"Gcp/terra-07381aa8","googleProject":"terra-07381aa8"},"patchInProgress":false},{"id":72404,"workspaceId":null,"runtimeName":"saturn-02636156-af51-4ca9-8cd8-c03a196c6fc7","googleProject":"terra-07381aa8","cloudContext":{"cloudProvider":"GCP","cloudResource":"terra-07381aa8"},"auditInfo":{"creator":"mcnatt@broadinstitute.org","createdDate":"2023-01-03T22:18:44.141960Z","destroyedDate":null,"dateAccessed":"2023-01-03T22:54:06.258843Z"},"runtimeConfig":{"machineType":"n1-standard-1","persistentDiskId":18662,"cloudService":"GCE","bootDiskSize":120,"zone":"us-central1-a","gpuConfig":null},"proxyUrl":"https://notebooks.firecloud.org/proxy/terra-07381aa8/saturn-02636156-af51-4ca9-8cd8-c03a196c6fc7/jupyter","status":"Stopped","labels":{"saturnWorkspaceNamespace":"qi-new-12-14","saturn-iframe-extension":"https://app.terra.bio/jupyter-iframe-extension.js","creator":"mcnatt@broadinstitute.org","clusterServiceAccount":"pet-266964691988800639cb9@terra-07381aa8.iam.gserviceaccount.com","saturnAutoCreated":"true","clusterName":"saturn-02636156-af51-4ca9-8cd8-c03a196c6fc7","saturnWorkspaceName":"qi-new-12-14","saturnVersion":"6","tool":"Jupyter","runtimeName":"saturn-02636156-af51-4ca9-8cd8-c03a196c6fc7","cloudContext":"Gcp/terra-07381aa8","googleProject":"terra-07381aa8"},"patchInProgress":false}]%
```


So I think there is sth wrong with support of `creator=<email>` syntax..altho in some tests I saw on dev, it worked, which is why I thought things are working as expected

**Comment by n8landolt** [comment]
Valid point @cahrens. My mistake for approving so quickly.

I'll get on the testing.

File: src/pages/workspaces/workspace/WorkspaceContainer.js
Comment (nawatts) [inline src/pages/workspaces/workspace/WorkspaceContainer.js]: Does the other branch of this condition also need to use `role: 'creator'` instead of `creator: getUser().email`?

**Review by nawatts — APPROVED** [review]

**Review by n8landolt — APPROVED** [review]
Seeing if this will fix the blocked merge even though there's two approvers.

**Review by cahrens — APPROVED** [review]

**Review by aherbst-broad — APPROVED** [review]

**Pull request merged**
